File: upack/PackageMetadata.cs
using System.Runtime.Serialization;

namespace Inedo.ProGet.UPack
{
    [DataContract]
    internal sealed class PackageMetadata
    {
        [DataMember(IsRequired = false, Name = "group")]
        public string Group { get; set; }
        [DataMember(IsRequired = true, Name = "name")]
        public string Name { get; set; }
        [DataMember(IsRequired = true, Name = "version")]
        public string Version { get; set; }
        [DataMember(IsRequired = false, Name = "title")]
        public string Title { get; set; }
        [DataMember(IsRequired = false, Name = "description")]
        public string Description { get; set; }
        [DataMember(IsRequired = false, Name = "icon")]
        public string IconUrl { get; set; }
        [DataMember(IsRequired = false, Name = "dependencies")]
        public string[] Dependencies { get; set; }

        public string BareVersion
		{
            get
            {
                var packageVersion = UniversalPackageVersion.TryParse(Version);
                return packageVersion != null
                    ? $"{packageVersion.Major}.{packageVersion.Minor}.{packageVersion.Patch}"
                    : Version;
            }
        }
    }
}

File: package_metadata.go
package main

type PackageMetadata struct {
	Group        string   `json:"group,omitempty"`
	Name         string   `json:"name"`
	Version      string   `json:"version"`
	Title        string   `json:"title,omitempty"`
	Description  string   `json:"description,omitempty"`
	IconURL      string   `json:"icon,omitempty"`
	Dependencies []string `json:"dependencies,omitempty"`
}

func (meta PackageMetadata) BareVersion() string {
	v, err := ParseUniversalPackageVersion(meta.Version)

	if err == nil {
		v.Prerelease = ""
		v.Build = ""
		return v.String()
	}

	return meta.Version
}

func (meta PackageMetadata) groupAndName() string {
	if meta.Group != "" {
		return meta.Group + "/" + meta.Name
	}
	return meta.Name
}
